compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -307,10 +307,42 @@ pub fn from_fn_attrs<'ll, 'tcx>(
         to_add.push(llvm::CreateAttrStringValue(cx.llcx, "branch-target-enforcement", "false"));
     }
     if codegen_fn_attrs.flags.contains(CodegenFnAttrFlags::ALLOCATOR) {
+        if llvm_util::get_version() >= (15, 0, 0) {
+            to_add.push(llvm::CreateAttrStringValue(cx.llcx, "alloc-family", "__rust_alloc"));
+            // apply to argument place instead of function
+            let alloc_align = AttributeKind::AllocAlign.create_attr(cx.llcx);
+            attributes::apply_to_llfn(llfn, AttributePlace::Argument(1), &[alloc_align]);
+        }
+        to_add.push(llvm::CreateAllocSizeAttr(cx.llcx, 0));
```

Comment:
Might as well put this into the LLVM 15 branch as well? Don't think there's much point to emitting it before that, even if it exists.

Reply:
Sure.